sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -661,16 +661,20 @@ def create_transform(self, type, args, yaml_create_transform):
     return self._transform_factories[type](yaml_create_transform, **args)
 
 
+def get_default_sql_provider():
```

Comment:
Could we add a follow on PR adding a comment mentioning that this function is special? Otherwise I worry that folks will accidentally overwrite it without understanding the implications.

@robertwb in general it would probably be better if we could resolve conversations before merging

Reply:
> Could we add a follow on PR adding a comment mentioning that this function is special? Otherwise I worry that folks will accidentally overwrite it without understanding the implications.

I added https://github.com/apache/beam/pull/34409

Reply:
Yes, sorry I didn't see the comment while I was reviewing it, and merging it immediately after. Thanks for following up.